repository.go

```diff
@@ -223,29 +242,14 @@ func (r *repository) fetch() error {
 
 	r.Lock()
 	r.etag = resp.Header.Get("Etag")
-	r.segments = featureResp.SegmentsMap()
-	r.options.storage.Reset(featureResp.FeatureMap(), true)
+	r.saveState(&featureResp)
 	r.successfulFetch()
 	r.Unlock()
 	return nil
 }
 
-// updateStorageWithDelta updates the storage with delta changes in a thread-safe manner
-func (r *repository) updateStorageWithDelta(features map[string]*api.Feature, segments map[int][]api.Constraint) error {
```

Comment:
This now goes via the one and only one way to update state. I think this method only really exists because the delta processor and the repository are strangely coupled at the moment